src/format/parse.rs

```diff
@@ -216,7 +216,11 @@ pub(crate) fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseRes
     }
 
     let offset = try_consume!(scan::timezone_offset(s, |s| scan::char(s, b':'), true, false, true));
-    if offset <= -86_400 || offset >= 86_400 {
+    // This range check is similar to the one in `FixedOffset::east_opt`, so it would be redundant.
+    // But it is possible to read the offset directly from `Parsed`. We want to only successfully
+    // populate `Parsed` if the input is fully valid RFC 3339.
+    const MAX_OFFSET: i32 = 23 * 3600 + 59 * 60;
```

Comment:
I'd like for this to be two commits:

* One for using a constant and the reformulated `Range::contains()`
* One for changing the value from 86400 to 863400

I also think `23 * 3600 + 59 * 60` is a surprising way to express this? I think (24 * 60 - 1) * 60 might be clearer.

Reply:
> I also think `23 * 3600 + 59 * 60` is a surprising way to express this? I think (24 * 60 - 1) * 60 might be clearer.

It comes from the definition in RFC 3339 as copied in comment of this function:
```
// time-hour = 2DIGIT ; 00-23
// time-numoffset = ("+" / "-") time-hour ":" time-minute
// time-minute = 2DIGIT ; 00-59
// time-offset = "Z" / time-numoffset
```

Reply:
Right. (23 * 60 + 59) * 60 could also work, I guess, but I find the previous suggestion more intuitive. I'll leave it up to you.

Reply:
Done. Added 'Max for the hours field is `23`, and for the minutes field `59`.' as a comment.